### PR TITLE
fix(filter-field): Fixes and issue where filters are not immediately removed when clicking the clear all button

### LINKED ISSUE #176: Filter field: Filters are not immediately removed when clicking the clear all button
You have to click somewhere outside to trigger the removal of the filter field.
To me this look like a CD issue we have here. Maybe we have to trigger it by hand somewhere.

[See screencast](https://github.com/Dynatrace/barista/files/3874154/Filter.field._.Barista.Design.System.-.Google.Chrome.2019-11-21.11-08-01.mp4.zip)

### PR DESCRIPTION
**Pull request by thomaspink**
fix(filter-field): Fixes and issue where filters are not immediately removed when clicking the clear all button

Fixes #176

### CODE REVIEW

**Comment by tomheller** [comment]
PR looks good, but can we adde some tests that would test this behaviour?

File: components/filter-field/src/filter-field.ts [hunk 1]
```diff
@@ -618,7 +618,6 @@ export class DtFilterField<T> implements AfterViewInit, OnDestroy, OnChanges {
 
   /** @internal Clears all filters and switch to root def. */
   _clearAll(event: Event): void {
```
Comment (tomheller) [inline components/filter-field/src/filter-field.ts]: remove `event`, if we no longer need it

**Comment by thomaspink** [comment]
Sadly, this can not be tested via unit tests because the `event.stopPropagation()` stops zone(angular) from running CD - but in unit tests you trigger CD by hand which means that the clearAll button will alway work there (thats why no tests are failing).

Will try to come up with an e2e test

**Comment by tomheller** [comment]
I guess some e2e tests wont hurt the filter field anyways, let's see if we can add a test for to be on the save side.

**Pull request merged**